### PR TITLE
PR1992-SDK chnages

### PR DESCRIPTION
**Pull request by kapilkumar99**
modified files:-generateobjectfromschema, in order to reflect the  manual code changes, in apicontracts

### CODE REVIEW

File: lib/apicontracts.js [hunk 1]
```diff
@@ -1863,19 +1943,25 @@ class ListOfAUDetailsType {
 		logger.debug('Enter ListOfAUDetailsType constructor');
 		if(arguments.length == 1) {
 			if((obj != undefined) && (obj != null)) {
-				var auUpdateOrAuDeleteList = [];
-				obj.forEach(function(item) {auUpdateOrAuDeleteList.push(new String(item));}); 
-				this.setAuUpdateOrAuDelete(auUpdateOrAuDeleteList);
+				var auUpdateList = [];
+				obj.forEach(function(item){if(item.newCreditCard != null) {auUpdateList.push(new AuUpdateType(item));}});
```
Comment (gnongsie) [inline lib/apicontracts.js]: Split this up across multiple lines for readability

File: lib/apicontracts.js [hunk 2]
```diff
@@ -1863,19 +1943,25 @@ class ListOfAUDetailsType {
...
 		if(arguments.length == 1) {
 			if((obj != undefined) && (obj != null)) {
-				var auUpdateOrAuDeleteList = [];
-				obj.forEach(function(item) {auUpdateOrAuDeleteList.push(new String(item));}); 
-				this.setAuUpdateOrAuDelete(auUpdateOrAuDeleteList);
+				var auUpdateList = [];
+				obj.forEach(function(item){if(item.newCreditCard != null) {auUpdateList.push(new AuUpdateType(item));}}); 
+				this.setAuUpdate(auUpdateList);
+				var auDeleteList = [];
+				obj.forEach(function(item){if(item.creditCard != null) {auDeleteList.push(new AuDeleteType(item));}});
```
Comment (gnongsie) [inline lib/apicontracts.js]: Split this up across multiple lines for readability

File: lib/apicontracts.js [hunk 3]
```diff
@@ -2842,20 +3060,40 @@ class SubscriptionPaymentType {
...
+module.exports.SubsequentAuthInformation = SubsequentAuthInformation;
+
 class TokenMaskedType {
 	constructor(obj) {
 		logger.debug('Enter TokenMaskedType constructor');
 		if(arguments.length == 1) {
 			if(('tokenSource' in obj) && (obj.tokenSource != null)) { this.setTokenSource(obj.tokenSource); }
 			if(('tokenNumber' in obj) && (obj.tokenNumber != null)) { this.setTokenNumber(obj.tokenNumber); }
 			if(('expirationDate' in obj) && (obj.expirationDate != null)) { this.setExpirationDate(obj.expirationDate); }
-			if(('tokenRequestorId' in obj) && (obj.tokenRequestorId != null)) { this.setTokenRequestorId(obj.tokenRequestorId); }
```
Comment (gnongsie) [inline lib/apicontracts.js]: Why was this removed?

File: lib/apicontracts.js [hunk 4]
```diff
@@ -2842,20 +3060,40 @@ class SubscriptionPaymentType {
...
 		}
 		else {
 			this.setTokenSource(null);
 			this.setTokenNumber(null);
 			this.setExpirationDate(null);
-			this.setTokenRequestorId(null);
```
Comment (gnongsie) [inline lib/apicontracts.js]: Why was this removed?

File: lib/apicontracts.js [hunk 5]
```diff
@@ -2866,8 +3104,6 @@ class TokenMaskedType {
 	getTokenNumber() { if('tokenNumber' in this) {return this.tokenNumber;} }
 	setExpirationDate(p_expirationDate) { this.expirationDate = p_expirationDate; }
 	getExpirationDate() { if('expirationDate' in this) {return this.expirationDate;} }
-	setTokenRequestorId(p_tokenRequestorId) { this.tokenRequestorId = p_tokenRequestorId; }
-	getTokenRequestorId() { if('tokenRequestorId' in this) {return this.tokenRequestorId;} }
```
Comment (gnongsie) [inline lib/apicontracts.js]: Why were these lines removed?

File: lib/apicontracts.js [hunk 6]
```diff
@@ -4827,18 +5099,12 @@ class CreditCardType extends CreditCardSimpleType {
 			if(('cardCode' in obj) && (obj.cardCode != null)) { this.setCardCode(obj.cardCode); }
 			if(('isPaymentToken' in obj) && (obj.isPaymentToken != null)) { this.setIsPaymentToken(obj.isPaymentToken); }
 			if(('cryptogram' in obj) && (obj.cryptogram != null)) { this.setCryptogram(obj.cryptogram); }
-			if(('tokenRequestorName' in obj) && (obj.tokenRequestorName != null)) { this.setTokenRequestorName(obj.tokenRequestorName); }
-			if(('tokenRequestorId' in obj) && (obj.tokenRequestorId != null)) { this.setTokenRequestorId(obj.tokenRequestorId); }
-			if(('tokenRequestorEci' in obj) && (obj.tokenRequestorEci != null)) { this.setTokenRequestorEci(obj.tokenRequestorEci); }
```
Comment (gnongsie) [inline lib/apicontracts.js]: Why were these lines removed?

File: lib/apicontracts.js [hunk 7]
```diff
@@ -4827,18 +5099,12 @@ class CreditCardType extends CreditCardSimpleType {
...
 		}
 		else {
 			super();
 			this.setCardCode(null);
 			this.setIsPaymentToken(null);
 			this.setCryptogram(null);
-			this.setTokenRequestorName(null);
-			this.setTokenRequestorId(null);
-			this.setTokenRequestorEci(null);
```
Comment (gnongsie) [inline lib/apicontracts.js]: Why were these lines removed?

File: lib/apicontracts.js [hunk 8]
```diff
@@ -4849,12 +5115,6 @@ class CreditCardType extends CreditCardSimpleType {
 	getIsPaymentToken() { if('isPaymentToken' in this) {return this.isPaymentToken;} }
 	setCryptogram(p_cryptogram) { this.cryptogram = p_cryptogram; }
 	getCryptogram() { if('cryptogram' in this) {return this.cryptogram;} }
-	setTokenRequestorName(p_tokenRequestorName) { this.tokenRequestorName = p_tokenRequestorName; }
-	getTokenRequestorName() { if('tokenRequestorName' in this) {return this.tokenRequestorName;} }
-	setTokenRequestorId(p_tokenRequestorId) { this.tokenRequestorId = p_tokenRequestorId; }
-	getTokenRequestorId() { if('tokenRequestorId' in this) {return this.tokenRequestorId;} }
-	setTokenRequestorEci(p_tokenRequestorEci) { this.tokenRequestorEci = p_tokenRequestorEci; }
-	getTokenRequestorEci() { if('tokenRequestorEci' in this) {return this.tokenRequestorEci;} }
```
Comment (gnongsie) [inline lib/apicontracts.js]: Why were these lines removed?

File: scripts/generateObjectsFromSchema.js [hunk 9]
```diff
@@ -126,28 +126,92 @@ for(i=0;i<schema.typeInfos.length;i++) {
...
-				if(obj['localName'].startsWith('ArrayOf') || obj.propertyInfos.length == 1)
+				
+				// if(obj['localName'].startsWith('ArrayOf') || obj.propertyInfos.length == 1)
+					// propname = 'obj';
+				// var listname = name + 'List';
+
+				// copyCtor += tab.repeat(tabcount) + 'if((' + propname + ' != undefined) && (' +  propname + ' != null)) {' + newline;
+				// tabcount++;
+
+				// copyCtor += tab.repeat(tabcount) + 'var ' + listname + ' = [];' + newline;
```
Comment (gnongsie) [inline scripts/generateObjectsFromSchema.js]: Removed commented code.

File: scripts/generateObjectsFromSchema.js [hunk 10]
```diff
@@ -126,28 +126,92 @@ for(i=0;i<schema.typeInfos.length;i++) {
...
-				copyCtor += '(item));}); ' + newline;
-				copyCtor += tab.repeat(tabcount) + 'this.set' + name.charAt(0).toUpperCase() + name.slice(1) + '(' + listname + ');';
-				copyCtor += newline;
-				tabcount--;
-				copyCtor += tab.repeat(tabcount) + '}' + newline;
+				// copyCtor += '(item));}); ' + newline;
+				// copyCtor += tab.repeat(tabcount) + 'this.set' + name.charAt(0).toUpperCase() + name.slice(1) + '(' + listname + ');';
+				// copyCtor += newline;
+				// tabcount--;
+				// copyCtor += tab.repeat(tabcount) + '}' + newline;
```
Comment (gnongsie) [inline scripts/generateObjectsFromSchema.js]: Remove commented code.

File: lib/apicontracts.js (hunk 6 above)
Comment (vijayabraj) [reply lib/apicontracts.js]: These lines were added for ChasePay sdk changes. Did you make sure your fork was updated before you made a commit and push?

**Comment by gnongsie** [comment]
Review complete. Ready to be merged.

January 2019 Release
